firebase-storage/src/main/java/com/google/firebase/storage/GetMetadataTask.java

```diff
@@ -41,6 +41,9 @@
 
     this.mStorageRef = storageRef;
     this.mPendingResult = pendingResult;
+    if (storageRef.getRoot().getName().equals(storageRef.getName())) {
+      throw new IllegalArgumentException("Cannot get object metedata at the root of a bucket.");
+    }
```

Comment:
Can you make the error message something like: "getMetadata() is not supported at the root of the Storage bucket."

Reply:
Done